partiql-ast/src/main/java/org/partiql/ast/WindowFunctionType.java

```diff
@@ -0,0 +1,309 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.expr.Expr;
+import org.partiql.ast.expr.ExprWindowFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a window function type.
+ * @see ExprWindowFunction#getFunctionType()
+ * @deprecated This feature is experimental and is subject to change.
+ */
+@Deprecated
+public abstract class WindowFunctionType extends AstNode {
+
+    /**
+     * The RANK window function.
+     * @see ExprWindowFunction#getFunctionType()
+     * @deprecated This feature is experimental and is subject to change.
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    @Deprecated
+    public static final class Rank extends WindowFunctionType {
+        private int _type;
```

Comment:
nit: this kind of feels a lot like `AstEnum`. If we can't make `Rank` extend `AstEnum` as well (I think this is not permitted in java to extend multiple abstract classes), perhaps we can just have these as their own variant of `WindowFunctionType` since there's no shared properties. Like

```
public static final class Rank extends WindowFunctionType { ... }
public static final class DenseRank extends WindowFunctionType { ... }
public static final class PercentRank extends WindowFunctionType { ... }
```